test/extended/operators/olm.go

```diff
@@ -37,6 +38,32 @@ var _ = g.Describe("[Feature:Platform] OLM should", func() {
 			e2e.Logf("Successfully list %s", olmResources[i])
 		})
 	}
+
+	// OCP-24074 Check OLM resources API Version
+	// Author: jiazha@redhat.com
+	versionMap := map[string]string{
+		"operatorgroups":         "v1",
+		"packagemanifests":       "v1",
+		"catalogsources":         "v1alpha1",
+		"subscriptions":          "v1alpha1",
+		"installplans":           "v1alpha1",
+		"clusterserviceversions": "v1alpha1",
+	}
+
+	for k, v := range versionMap {
+		g.It("check API version: "+k, func() {
+			output, err := oc.AsAdmin().Run("get").Args(k, "-n", "openshift-operator-lifecycle-manager", "-o=jsonpath={.items[0].apiVersion}").Output()
```

Comment:
In addition to this, I think we should also check the [`versions` field](https://v1-14.docs.kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#create-a-customresourcedefinition) of its CRDs.

Reply:
Thanks! Add the CRD checking.